pamfixtrunc, pamfix, pamvalidate: add pages (#12406)
tldr-pages/tldr: pages/common/pamfix.md

# pamfix

> > Fix errors in Netpbm files.

Reviewer: ```suggestion
> Fix errors in PAM, PBM, PGM and PPM files.
```

> See also: `pamfile`, `pamvalidate`.
> More information: <https://netpbm.sourceforge.net/doc/pamfix.html>.

- Fix a Netpbm file that is missing its last part:

> `pamfix -truncate {{path/to/corrupted.pam}} > {{path/to/output.pam}}`

Reviewer: ```suggestion
`pamfix -truncate {{path/to/corrupted.pam|pbm|pgm|ppm}} > {{path/to/output.pam|pbm|pgm|ppm}}`
```

> `pamfix -truncate {{path/to/corrupted.[pam|pbm|pgm|ppm]}} > {{path/to/output.[pam|pbm|pgm|ppm]}}`

Reviewer: ```suggestion
`pamfix -truncate {{path/to/corrupted.ext} > {{path/to/output.ext}}`
```

> - Fix a Netpbm file where pixel values exceed the image's maxval by lowering the offending pixels' values:

Reviewer 2: ```suggestion
- Fix a Netpbm file where pixel values exceed the image's maximum value by lowering the offending pixels' values:
```

Author: "maxval" is a fixed term in the Netpbm format.

> - Fix a Netpbm file where pixel values exceed the image's maxval by lowering the offending pixels' values:

Reviewer: ```suggestion
- Fix a Netpbm file where pixel values exceed the image's `maxval` by lowering the offending pixels' values:
```

> `pamfix -clip {{path/to/corrupted.pam}} > {{path/to/output.pam}}`

Reviewer: ```suggestion
`pamfix -clip {{path/to/corrupted.pam|pbm|pgm|ppm}} > {{path/to/output.pam|pbm|pgm|ppm}}`
```

> `pamfix -clip {{path/to/corrupted.[pam|pbm|pgm|ppm]}} > {{path/to/output.[pam|pbm|pgm|ppm]}}`

Reviewer: ```suggestion
`pamfix -clip {{path/to/corrupted.ext}} > {{path/to/output.ext}}`
```

> - Fix a Netpbm file where pixel values exceed the image's maxval increasing the maxval:

Reviewer 2: ```suggestion
- Fix a Netpbm file where pixel values exceed the image's maximum value by increasing itl:
```

> `pamfix -changemaxval {{path/to/corrupted.pam}} > {{path/to/output.pam}}`

Reviewer: ```suggestion
`pamfix -changemaxval {{path/to/corrupted.pam|pbm|pgm|ppm}} > {{path/to/output.pam|pbm|pgm|ppm}}`
```

Author: ```suggestion
`pamfix -changemaxval {{path/to/corrupted.pam|pbm|pgm|ppm}} > {{path/to/output.pam|pbm|pgm|ppm}}`
```

I don'tthink tldr has this syntax. I'd rather to this.

Reviewer 2: > I don'tthink tldr has this syntax. I'd rather to this.

We use this in quite a few places as a replacement to `.ext` in placeholders. I was planning to document this in the style guide with https://github.com/tldr-pages/tldr/issues/12396.

Author: You are right, we can introduce this but we should document it.

> `pamfix -changemaxval {{path/to/corrupted.pam}} > {{path/to/output.pam}}`

Reviewer: ```suggestion
`pamfix -changemaxval {{path/to/corrupted.ext}} > {{path/to/output.ext}}`
```
